### ISSUE TITLE
Question about the choice of use_fast_accum in FP8 Training

### ISSUE THREAD

**Issue by haochengxi** [pytorch/ao#2377]
In the FP8 training setting, I noticed that `use_fast_accum` is only enabled during the forward pass:
[https://github.com/pytorch/ao/blob/6243040807b9ceee889a58cba8e68c5fc4e2ebd8/torchao/float8/config.py#L174](https://github.com/pytorch/ao/blob/6243040807b9ceee889a58cba8e68c5fc4e2ebd8/torchao/float8/config.py#L174)

When benchmarking performance, I've observed that enabling fast accumulation can lead to a 1.1x to 1.2x speedup. I'm curious—was there a specific reason or concern behind the decision to restrict `use_fast_accum` to only the forward pass?

![Image](https://github.com/user-attachments/assets/72305633-118a-4c1e-9993-8f8b1c5e553e)

**Comment by vkuzo** [comment]
Sorry about the late reply! The fast_accum flag controls the accumulation behavior of the gemm kernel, and enabling it can speed up the gemm at the cost of some accuracy. The default setting we have reflect a perf vs accuracy tradeoff that we have generally found optimal, but you can definitely try using fast_accum more liberally to bias more towards performace.

Note that the gradients are often more spiky in magnitude than activations and weights, so extra accuracy tends to be more important in the backward.
